db/transaction.go

```diff
@@ -0,0 +1,157 @@
+package db
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/attic-labs/noms/go/types"
+
+	"roci.dev/diff-server/kv"
+	nomsjson "roci.dev/diff-server/util/noms/json"
+	"roci.dev/diff-server/util/time"
+)
+
+var (
+	// ErrClosed is the error returned from operations on a Transaction when
+	// it has already been closed.
+	ErrClosed = errors.New("Transaction is closed")
+)
+
+type Transaction struct {
+	db     *DB
+	head   Commit
+	me     *kv.MapEditor
+	wrote  bool
+	closed bool
+	name   string
+	args   types.List
+}
+
+// Closed returns true when the database has been closed. A database becomes
+// closed after Committ or Close is called.
+func (tx *Transaction) Closed() bool {
+	return tx.closed
+}
+
+// Get returns the JSON encoded value at the given id of the database.
+func (tx *Transaction) Get(id string) ([]byte, error) {
```

Comment:
fwiw these args are variously called id, path, and key in different places. not sure if it is worth trying to standardize on one.

Reply:
*ducks*

I think we should probably standardize on `key` since this is a "key value database". Whoops. This is fine for now.

Reply:
And also because this is a cache and generally the lookup thingy in a cache is called _key_.

Reply:
I prefer `key` but I don't have a string preference. I agree that we should standardize since it does come up in `ScanItem`

Reply:
key is great, let's do it